v0/destinations/moengage/transform.js

```diff
@@ -0,0 +1,129 @@
+const btoa = require("btoa");
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  endpointEU,
+  endpointIND,
+  endpointUS
+} = require("./config");
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultRequestConfig,
+  flattenJson
+} = require("../../util");
+
+function responseBuilderSimple(message, category, destination) {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  const { apiId, region, apiKey } = destination.Config;
+  const response = defaultRequestConfig();
+  // check the region and which api end point should be used
+  switch (region) {
+    case "EU":
+      response.endpoint = `${endpointEU[category.type]}${apiId}`;
+      break;
+    case "US":
+      response.endpoint = `${endpointUS[category.type]}${apiId}`;
+      break;
+    case "IND":
+      response.endpoint = `${endpointIND[category.type]}${apiId}`;
+      break;
+    default:
+      throw new Error("The region is not valid");
+  }
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    "MOE-APPKEY": apiId,
+    Authorization: `Basic ${btoa(`${apiId}:${apiKey}`)}` // Basic Authentication encodes a 'username:password' using base64 and prepends it with the string 'Basic '.
+  };
+  response.userId = message.anonymousId || message.userId;
+  if (payload) {
+    switch (category.type) {
+      case "identify":
+        payload.type = "customer";
+        // https://docs.moengage.com/docs/data-import-apis#user-api
+        payload.attributes = constructPayload(
+          message,
+          MAPPING_CONFIG[CONFIG_CATEGORIES.IDENTIFY_ATTR.name]
+        );
+        payload.attributes = flattenJson(payload.attributes); // nested attributes are not shown on dashboard of moengage so it is falttened
+        break;
+      case "device":
+        payload.type = "device";
+        // https://docs.moengage.com/docs/data-import-apis#device-api
+        payload.attributes = constructPayload(
+          message,
+          MAPPING_CONFIG[CONFIG_CATEGORIES.DEVICE_ATTR.name]
+        );
+        payload.attributes = flattenJson(payload.attributes); // nested attributes are not shown on dashboard of moengage so it is falttened
```

Comment:
Move the comments one line above. Line length ideally should be at 80 characters

Reply:
done